[Site Isolation] Fix page zoom

charliewolfe opened the pull request:
#### e231b8ad73c079d0e8b75fbdb7327c259e2eae12
<pre>
[Site Isolation] Fix page zoom
<a href="https://bugs.webkit.org/show_bug.cgi?id=273257">https://bugs.webkit.org/show_bug.cgi?id=273257</a>
<a href="https://rdar.apple.com/127061418">rdar://127061418</a>

Reviewed by Alex Christensen.

Send the zoom values to each web process and update zoom from the root frames.

* LayoutTests/http/tests/site-isolation/page-zoom-expected.html: Added.
* LayoutTests/http/tests/site-isolation/page-zoom.html: Added.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::setPageZoomFactor):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setPageZoomFactor):

Canonical link: <a href="https://commits.webkit.org/278002@main">https://commits.webkit.org/278002@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b4aa757658d808b3ed75a7d2b2aff571e1c29e92

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49168 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28449 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52192 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51966 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45268 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34441 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26043 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40184 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51008 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26011 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42379 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21297 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23465 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43556 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7492 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45393 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44061 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53877 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24247 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20462 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/47501 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25524 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42584 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46486 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26351 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7052 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25245 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (changes requested):
Could you mark your new radar a duplicate of rdar://117912896 ?
Also, the test compares site isolation enabled vs disabled, which is fine for now but we will lose test coverage when we enable site isolation.  Could you instead make a test that constructs what the page should look like after it is zoomed without using zoomPageIn?

achristensen07 reviewed (approved):
Same origin.  Indeed.

charliewolfe committed:
[Site Isolation] Fix page zoom
https://bugs.webkit.org/show_bug.cgi?id=273257
rdar://127061418

Reviewed by Alex Christensen.

Send the zoom values to each web process and update zoom from the root frames.

* LayoutTests/http/tests/site-isolation/page-zoom-expected.html: Added.
* LayoutTests/http/tests/site-isolation/page-zoom.html: Added.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::setPageZoomFactor):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setPageZoomFactor):

Canonical link: https://commits.webkit.org/278002@main
